### PR TITLE
Refactor: Snap State and Account Static Classes

### LINKED ISSUE #77: Class refactor
### Problem

There is currently a mix of object / functional programming.

### Solution

Refactor to use classes, using static classes/methods where appropriate for 'utility' like functions. 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by poi-son-ivy**
**Description**:
This PR refactors the snap/ directory into State, Account, and a HederaClientFactory static class with associated reference changes and tests.

**Related issue(s)**:

Fixes #77, #78

**Notes for reviewer**:
Tests are currently constrained by the snap global variable, this will be tracked as a separate issue and fixed if possible, or mitigated with functional tests. 

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts [hunk 1]
```diff
@@ -114,6 +114,7 @@ export async function transferCrypto(
       const txNumber = transfers.indexOf(transfer) + 1;
       panelToShow.push(text(`Transaction #${txNumber}`));
       panelToShow.push(divider());
+      panelToShow.push(divider());
```
Comment (kpachhai) [inline packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts:117]: This was removed on the latest commit on wallet-snap branch but it seems this PR is bringing it back. Let's make sure to remove this from the PR

File: packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts [hunk 2]
```diff
@@ -136,6 +137,7 @@ export async function transferCrypto(
         panelToShow.push(text(`Owner Account Id: ${transfer.from as string}`));
       }
       panelToShow.push(text(`Asset Type: ${transfer.assetType}`));
+      panelToShow.push(divider());
```
Comment (kpachhai) [inline packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts:140]: This was removed on the latest commit on wallet-snap branch but it seems this PR is bringing it back. Let's make sure to remove this from the PR

File: packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts (hunk 1 above)
Comment (poi-son-ivy) [reply packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts]: fixed!

File: packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts (hunk 2 above)
Comment (poi-son-ivy) [reply packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts]: fixed!

File: packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts [hunk 3]
```diff
@@ -168,18 +170,14 @@ export async function transferCrypto(
             ),
           );
         }
-
-        let assetId = transfer.assetId as string;
-        let nftSerialNumber = '';
```
[the remark this answers is not in the document]
Comment (poi-son-ivy) [reply packages/hedera-wallet-snap/packages/snap/src/rpc/transactions/transferCrypto.ts]: fixed!

**Review by kpachhai — APPROVED** [review]

**Review by kpachhai — APPROVED** [review]

**Pull request merged**